pkg/proxy/iptables/proxier.go

```diff
@@ -652,9 +634,9 @@ func updateServiceMap(
 		changes.Lock()
 		defer changes.Unlock()
 		for _, change := range changes.items {
-			mergeSyncRequired, existingPorts := serviceMap.mergeService(change.current)
-			unmergeSyncRequired := serviceMap.unmergeService(change.previous, existingPorts, staleServices)
-			syncRequired = syncRequired || mergeSyncRequired || unmergeSyncRequired
+			existingPorts := serviceMap.merge(change.current)
```

Comment:
I wanted to make this change effectively a no-op.
The problem is that for services we are keeping track of open ports and if the port was open before the change and is still open, we don't want to perform any operations for them.
With endpoints, we don't have such situation - we just want to update the state.

The easiest way to do it that I came up with is what I did. But if you have any other suggestions, I'm happy to change it in a followup PR.